c_src/erl_optics.c

```diff
@@ -511,13 +546,18 @@ static ERL_NIF_TERM eo_optics_poll(
     if (!optics) return ERROR("get_optics");
     if (!poller) return ERROR("poller_allocation_error");
 
-    ctx->env = env;
-    ctx->map = enif_make_new_map(env);
-
+    if (erlang_poller){
```

Comment:
Another minor detail: I'm seeing inconsistencies in the formatting and usually my solution for this is to just use something like clang-format to uniformize everything.

Note that consistent formatting makes things like searching code easier and generally makes things easier to read so it's a nice to have.